src/components/link-preview/link-preview.test.tsx

```diff
@@ -0,0 +1,91 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import LinkPreview from "./link-preview.component";
+
+test("renders with an anchor root element and hrf when the `url` prop is passed", () => {
+  render(<LinkPreview url="foo" />);
+  const linkPreviewWrapper = screen.getByRole("link");
+
+  expect(linkPreviewWrapper).toHaveAttribute("href", "foo");
+  expect(linkPreviewWrapper).toBeVisible();
+});
+
+test("renders with visible link text, via the `url` prop", () => {
+  render(<LinkPreview url="foo" />);
+  const linkText = screen.getByText("foo");
+
+  expect(linkText).toBeVisible();
+});
+
+test("renders with the schema removed in the visible link text, via the `url` prop if a schema is present", () => {
+  render(<LinkPreview url="https://foo" />);
+  const linkText = screen.getByText("foo");
+
+  expect(linkText).toBeVisible();
+});
+
+test('renders with a div root element, when the `as` prop is passed as "div"', () => {
+  render(<LinkPreview as="div" />);
+  const linkPreviewWrapper = screen.getByTestId("link-preview");
+  expect(linkPreviewWrapper.tagName).toBe("DIV");
```

Comment:
**suggestion/nitpick:** The first three tests have a line break before the `expect` assertion. I'm happy with either but make it consistent.